src/parse.rs

```diff
@@ -1,29 +1,62 @@
+use std::clone::Clone;
 use std::collections::HashSet;
 use std::path::Path;
 
+use cached::proc_macro::cached;
+use cached::UnboundCache;
 use html5ever::parse_document;
 use html5ever::rcdom::{Handle, NodeData, RcDom};
 use html5ever::tendril::TendrilSink;
 use log::{debug, info};
 use url::Url;
 
 /// Parse the html file at the provided path and check the availablility of all links in it.
-pub fn parse_html_file(root_dir: &Path, path: &Path) -> HashSet<Url> {
+pub fn parse_html_file(root_dir: &Path, path: &Path) -> (HashSet<Url>, HashSet<String>) {
     info!("Checking doc page at {}", path.display());
+    links_and_targets(root_dir, path)
+}
+
+#[cached(
+    type = "UnboundCache<String, (HashSet<Url>, HashSet<String>)>",
+    create = "{ UnboundCache::with_capacity(100) }",
+    convert = r#"{ format!("{}", path.clone().to_str().expect("Unable to convert path to str")) }"#
+)]
```

Comment:
This seems really complicated for what deadlinks is doing. Is there any reason to not just parse the file once and do all the checking at the same time?

Reply:
Hopefully I've explained this in the comment above. Ping me if not.